src/core/thread/link_quality.cpp

```diff
@@ -140,6 +140,32 @@ RssAverager::InfoString RssAverager::ToString(void) const
     return string;
 }
 
+void LqiAverager::Reset(void)
+{
+    mCount   = 0;
+    mAverage = 0;
+}
+
+void LqiAverager::Add(uint8_t aLqi)
+{
+    uint16_t oldAverage = mAverage;
+
+    mCount++;
```

Comment:
```suggestion
    mCount += (mCount < (1 << kCoeffBitShift));
```
should stop increase `mCount` if it reaches `1 << kCoeffBitShift` ? So, `mCount` will always be `1 << kCoeffBitShift` after it reaches this value.

Reply:
The count could be greater than `(1 << kCoeffBitShift)`. When count is greater than `(1 << kCoeffBitShift)`, the new average should be `7/8 * old value + 1/8 * new value`. Note that this coefficient(1/8) is fixed here.

Reply:
We should increase `mCount` to 8 (1 << kCoeffBitShift) and stop increasing it anymore. Otherwhise `mCount` will overflow and become 0 ~ 7 again.

Reply:
In `LqiAverager`, `mCount` has 8 bits and it's also intended to record the **Psdu count** for link metrics.

Reply:
Even so , we can have
```
count = OT_MIN(mCount, 8)
```

Then, use `count` in the formula.

Reply:
Make sense. I've updated. Thanks.